core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -230,7 +230,7 @@ public ActualProperties visitAssignUniqueId(AssignUniqueId node, List<ActualProp
             }
 
             return ActualProperties.builderFrom(properties)
-                    .global(partitionedOn(ARBITRARY_DISTRIBUTION, ImmutableList.of(node.getIdColumn()), Optional.empty()))
+                    .global(partitionedOn(ARBITRARY_DISTRIBUTION, ImmutableList.of(node.getIdColumn()), Optional.of(ImmutableList.of(node.getIdColumn()))))
```

Comment:
from cmt msg

> The uniqueId generated is based on StageId and TaskId so we could
specify stream partitioning for AssignUniqueId

I think that it doesn't matter how the ID is generated.
All we need to know is that the idColumn is populated with a value unique for every row.
If it's unique, it's also partitioned (both node-level and stream-level).

please update the commit message.